### ISSUE TITLE
Dask.array deepcopy does not preserve masking since 0.18.2

### ISSUE THREAD

**Issue by pp-mo** [dask/dask#3848]
If 'a' is a dask array wrapped around a numpy masked array (with "from _array(..., asarray=False)",
then the result of "copy.deepcopy(a).compute()" is now (wrongly) **not** a masked array,

Though the original "a.compute()", and even "copy.copy(a)" are correct.

Example code to show :
```
import dask.array as da
import numpy.ma as ma
import copy
t = ma.masked_array([1, 2], mask=[0, 1])
a = da.from_array(t, chunks=t.shape, asarray=False)
print(copy.deepcopy(a).compute())
```

For example, in dask 0.18.1
```
>>> import dask
>>> import dask.array as da
>>> import numpy.ma as ma
>>> import copy
>>> t = ma.masked_array([1, 2], mask=[0, 1])
>>> a = da.from_array(t, chunks=t.shape, asarray=False)
>>> print(dask.__version__)
0.18.1
>>> print(copy.copy(a).compute())
[1 --]
>>> print(copy.deepcopy(a).compute())
[1 --]
>>> 
```

But in 0.18.2 ...
```
>>> import dask
>>> import dask.array as da
>>> import numpy.ma as ma
>>> import copy
>>> 
>>> t = ma.masked_array([1, 2], mask=[0, 1])
>>> a = da.from_array(t, chunks=t.shape, asarray=False)
>>> 
>>> print(dask.__version__)
0.18.2
>>> print(copy.copy(a).compute())
[1 --]
>>> print(copy.deepcopy(a).compute())
[1 2]
>>> 
```

**Comment by pp-mo** [comment]
This has caused us to pin dask in [iris](https://github.com/SciTools/iris) :cry: 
as it breaks various unit tests for us.

I suspect the change [here](https://github.com/dask/dask/pull/3754/files#r207583443), but I don't know really enough to suggest a fix ... but surely this did not respect the comment anyway ?

**Comment by mrocklin** [comment]
cc @jcrist

**Comment by jcrist** [comment]
Looks like the `np.copy` in the `map_blocks` should be replaced with `dask.utils.M.copy` (which would call the `.copy` method on the array).

**Comment by mrocklin** [comment]
@pp-mo it looks like there is a proposed fix to your problem in #3852 .  Do you have time to take a look?
